extmod/uasyncio/lock.py

```diff
@@ -0,0 +1,48 @@
+# MicroPython uasyncio module
+# MIT license; Copyright (c) 2019-2020 Damien P. George
+
+from . import core
+
+# Lock class for primitive mutex capability
+class Lock:
+    def __init__(self):
+        self.state = 0  # 0=unlocked; 1=unlocked but waiting task pending resume; 2=locked
+        self.waiting = core.TaskQueue()  # Queue of Tasks waiting to acquire this Lock
+
+    def locked(self):
+        return self.state == 2
```

Comment:
Shouldn't this be `!= 0`? If `state == 1` the lock is effectively acquired by some task it just happens that that task hasn't resumed yet. I would expect: `locked()==True` <=> calling `acquire()` would block.

Reply:
No this is actually expected behaviour and how it is in CPython. If you call "lock.release()" then the lock is released. Otherwise this would fail:

```
async with lock:
    pass
if lock.locked():
    print("Why is lock still locked although just released??") 
```
You can't keep a lock locked only because a different task is going to acquire it.

Reply:
I see, could be argued either way I guess. E.g. what is the practical use of your sample code? What I was looking for was:
```
if not lock.locked(): # avoid blocking
    async with lock:
        do_something()
```
but since CPython works one way...

Reply:
Yes I understand your point and that my example is generally quite useless except for checking, that a lock *really* got released (which is stupid, i know). 
Also logic tells that if you call lock.release() and then check if the lock got released, it has to be released.. 
However, since this is CPython behaviour anyway..

What is the use-case of not waiting for a lock if it has other tasks waiting for it? If you are afraid of waiting too long, you can use asyncio.wait_for?

Reply:
Let's handle this after merge.